### ISSUE TITLE
Constant should probably use postgresql-simple's ToField

### ISSUE THREAD

**Issue by tomjaguarpaw** [tomjaguarpaw/haskell-opaleye#214]
No point rolling our own when this already exists:

https://hackage.haskell.org/package/postgresql-simple-0.5.2.1/docs/Database-PostgreSQL-Simple-ToField.html

**Comment by saurabhnanda** [comment]
@tomjaguarpaw what needs to be done here? I'll try a shot at writing the PR. Do we want a helper function on the following lines:

```
instance Default Constant haskType (Column pgType) where
  def = toFieldDefaultConstant
```

although I hate the name - `toFieldDefaultConstant`

Related to #281

**Comment by tomjaguarpaw** [comment]
Yes, exactly.  Feel free to choose whatever name you think is good.  I guess we need to use `Show` on the action to get the `String` to insert into the query, but I'm not sure.

**Comment by tomjaguarpaw** [comment]
`ToField` ultimately uses

    buildQuery :: Connection -> Query -> ByteString -> [Action] -> IO Builder

and we don't want `IO` in `Constant`, so we won't do anything more about this now.
